spring-context/src/main/java/org/springframework/context/annotation/ConfigurationClassParser.java

```diff
@@ -475,6 +481,32 @@ private void processPropertySource(AnnotationAttributes propertySource) throws I
 		}
 	}
 
+	/**
+	 * Parse locations and expand those with wildcard
+	 */
+	private String[] getWildcardLocations(String[] locations) throws IOException {
```

Comment:
If we were to introduce support for wildcards, we would need to use `PathMatchingResourcePatternResolver` instead of implementing a custom method like you have done here.